### ISSUE TITLE
No Dimming on "Smart-by-Bond" Devices

### ISSUE THREAD

**Issue by Silmar-99** [aarons22/homebridge-bond#104]
Hey Aaron,

I'm bbbaaack :)

Now that I have all my fans hooked and working as they should (thanks again for investigating this one for me), and since I can control independently the fan and the light fixture, I decided to re-assign my Philips Hue bulbs to somewhere else and replace them by "regular" LED bulbs.

However, I cannot seems to be able to get the "Dim Switch" you mention in the setup of your plugin.

Dimming is activated, working with the physical remote and from the Bond's app, but nada when it comes to HomeKit.

It sounds like it is not recognized as a "dimmable" device by the plugin.

Any idea on how to fix that and make it works?

As always, thanks for your time investigating this new issue.

Best regards,

**Comment by aarons22** [comment]
Hey @Silmar-99  - Welcome back! 😄 

Interesting, I wonder why it wouldn't report dimming in the same manner. If you are able to provide logs (specifically when in debug mode), that is always the most helpful. Alternatively, you can provide your cachedAccessories file which should contain all of the info I need!

**Comment by Silmar-99** [comment]
Hey Aaron!

Here the log...

[10/6/2020, 6:54:39 AM] [Bond] Request (185d3f70ddcec000) [get http://192.168.0.118/v2/devices]
[10/6/2020, 6:54:39 AM] [Bond] Request (185d3f70ddcf8000) [get http://192.168.0.118/v2/sys/version]
[10/6/2020, 6:54:39 AM] [Bond] Request (185d3f70ddcfc000) [get http://192.168.0.119/v2/devices]
[10/6/2020, 6:54:39 AM] [Bond] Request (185d3f70ddcfc001) [get http://192.168.0.119/v2/sys/version]
[10/6/2020, 6:54:39 AM] [Bond] Request (185d3f70ddd00000) [get http://192.168.0.120/v2/devices]
[10/6/2020, 6:54:39 AM] [Bond] Request (185d3f70ddd00001) [get http://192.168.0.120/v2/sys/version]
…
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70ddcfc000) [get http://192.168.0.119/v2/devices] - {"1":{"_":"4725c8a7"},"_":"9eddafa3"}
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70ddcfc001) [get http://192.168.0.119/v2/sys/version] - {"target":"breck-northstar","fw_ver":"v2.14.7","fw_date":"Mon Sep 14 19:00:53 UTC 2020","uptime_s":494962,"make":"Unbranded","model":"MR196W","branding_profile":"KOF_MR196W","bondid":"KNKSADE42861","upgrade_http":true,"api":2,"_":"3a4ef21c"}
[10/6/2020, 6:54:40 AM] [Bond] 
****** Bond Info *******
 bondId: KNKSADE42861
 FW: v2.14.7
 API: v2
 Make: MR196W
 Model: MR196W
************************
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70ddcec000) [get http://192.168.0.118/v2/devices] - {"1":{"_":"dd93a270"},"_":"dd28ddd2"}
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70ddd00000) [get http://192.168.0.120/v2/devices] - {"1":{"_":"09d3f8e8"},"_":"46c85b51"}
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70ddcf8000) [get http://192.168.0.118/v2/sys/version] - {"target":"breck-northstar","fw_ver":"v2.14.7","fw_date":"Mon Sep 14 19:00:53 UTC 2020","uptime_s":495472,"make":"Chungear Industrial Co., Ltd.","model":"MR196W","branding_profile":"KOF_KNKSA","bondid":"KNKSACG91580","upgrade_http":true,"api":2,"_":"43c57deb"}
[10/6/2020, 6:54:40 AM] [Bond] 
****** Bond Info *******
 bondId: KNKSACG91580
 FW: v2.14.7
 API: v2
 Make: MR196W
 Model: MR196W
************************
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70ddd00001) [get http://192.168.0.120/v2/sys/version] - {"target":"breck-northstar","fw_ver":"v2.14.7","fw_date":"Mon Sep 14 19:00:53 UTC 2020","uptime_s":493938,"make":"Unbranded","model":"MR196W","branding_profile":"KOF_MR196W","bondid":"KNKSADE41829","upgrade_http":true,"api":2,"_":"16fd4db2"}
[10/6/2020, 6:54:40 AM] [Bond] 
****** Bond Info *******
 bondId: KNKSADE41829
 FW: v2.14.7
 API: v2
 Make: MR196W
 Model: MR196W
************************
[10/6/2020, 6:54:40 AM] [Bond] Config: {"bonds":[{"ip_address":"192.168.0.118","token":"8f0ec3012675e6c2"},{"ip_address":"192.168.0.119","token":"bb5c9f6bade6b290"},{"ip_address":"192.168.0.120","token":"f2bd369c0b05ead7"}],"platform":"Bond","include_dimmer":true,"include_toggle_state":false,"fan_speed_values":false}
[10/6/2020, 6:54:40 AM] [Bond] 3 cached accessories were loaded
[10/6/2020, 6:54:40 AM] [Bond] Getting devices for this Bond (KNKSACG91580)...
[10/6/2020, 6:54:40 AM] [Bond] 1 devices were found on this Bond (KNKSACG91580).
[10/6/2020, 6:54:40 AM] [Bond] No new devices to add for this Bond (KNKSACG91580).
[10/6/2020, 6:54:40 AM] [Bond] Getting devices for this Bond (KNKSADE42861)...
[10/6/2020, 6:54:40 AM] [Bond] 1 devices were found on this Bond (KNKSADE42861).
[10/6/2020, 6:54:40 AM] [Bond] No new devices to add for this Bond (KNKSADE42861).
[10/6/2020, 6:54:40 AM] [Bond] Getting devices for this Bond (KNKSADE41829)...
[10/6/2020, 6:54:40 AM] [Bond] 1 devices were found on this Bond (KNKSADE41829).
[10/6/2020, 6:54:40 AM] [Bond] No new devices to add for this Bond (KNKSADE41829).
[10/6/2020, 6:54:40 AM] [Bond] UDP message sent to 192.168.0.118:30007
[10/6/2020, 6:54:40 AM] [Bond] UDP message sent to 192.168.0.119:30007
[10/6/2020, 6:54:40 AM] [Bond] UDP message sent to 192.168.0.120:30007
[10/6/2020, 6:54:40 AM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861"}
[10/6/2020, 6:54:40 AM] [myQ] myQ API: Successfully connected to the myQ API.
[10/6/2020, 6:54:40 AM] [Bond] UDP Message received from 192.168.0.120:30007 - {"B":"KNKSADE41829"}
[10/6/2020, 6:54:40 AM] [Bond] UDP Message received from 192.168.0.118:30007 - {"B":"KNKSACG91580"}
[10/6/2020, 6:54:40 AM] [Bond] Configuring Accessory: Master Bedroom MBR Fan
[10/6/2020, 6:54:40 AM] [Bond] Request (185d3f70dec98000) [get http://192.168.0.118/v2/devices/1/state]
[10/6/2020, 6:54:40 AM] [Bond] Configuring Accessory: Dining Room DR Fan
[10/6/2020, 6:54:40 AM] [Bond] Request (185d3f70deca8000) [get http://192.168.0.119/v2/devices/1/state]
[10/6/2020, 6:54:40 AM] [Bond] Configuring Accessory: Living Room LV Fan
[10/6/2020, 6:54:40 AM] [Bond] Request (185d3f70decb4000) [get http://192.168.0.120/v2/devices/1/state]
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70dec98000) [get http://192.168.0.118/v2/devices/1/state] - {"power":1,"speed":1,"light":1,"brightness":100,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"4684bfc5"}
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70deca8000) [get http://192.168.0.119/v2/devices/1/state] - {"power":0,"speed":2,"light":0,"brightness":100,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"e34e6754"}
[10/6/2020, 6:54:40 AM] [Bond] Response (185d3f70decb4000) [get http://192.168.0.120/v2/devices/1/state] - {"power":0,"speed":2,"light":0,"brightness":100,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"99987699"}

**Comment by Silmar-99** [comment]
And the link to download the file, since the format is not supported...
https://www.dropbox.com/s/fp6ktt3uh9uaoz1/cachedAccessories?dl=1

**Comment by aarons22** [comment]
Hey @Silmar-99 - After looking into it, it seems like you may have added dimmer functionality in Bond _after_ you added your fans? With the way the plugin works, it gets actions when initially adding the device, and then those actions get cached moving forward. Based on your cached accessories file, your device has no `StartDimmer` action, which is required for the dimming functionality.

So, to refresh the actions, you will need to either remove the Bond accessories from your cached accessories file, or delete the file completely (I know the later might break other plugins, which isn't ideal).

**Comment by Silmar-99** [comment]
Hey Aaron,
Apologies as I was out of town last week.
So I did remove one device for testing (in both the cache and the plugin), in order to test the feature.
Once removed (and re-detected), the device does show both buttons and the possibility to slide the light cursor but, it is not doing anything...
The light turn on and off when it reach the bottom of the cursor track but, once on, the intensity of the light do not change...
Any idea?

**Comment by Silmar-99** [comment]
Good afternoon Aaron,
Any update on the above by any chance?

**Comment by aarons22** [comment]
Hey @Silmar-99 - sorry for the delay. My ability to work on this seems to come in waves.

Can you provide logs for the scenario you tested on November 9?

**Comment by Silmar-99** [comment]
Hey Aaron,

Here you go...

[12/14/2020, 8:26:58 PM] [Bond] Request (185d990e38e7c000) [get http://192.168.0.118/v2/devices]
[12/14/2020, 8:26:58 PM] [Bond] Request (185d990e38e88000) [get http://192.168.0.118/v2/sys/version]
[12/14/2020, 8:26:58 PM] [Bond] Request (185d990e38e8c000) [get http://192.168.0.119/v2/devices]
[12/14/2020, 8:26:58 PM] [Bond] Request (185d990e38e90000) [get http://192.168.0.119/v2/sys/version]
[12/14/2020, 8:26:58 PM] [Bond] Request (185d990e38e90001) [get http://192.168.0.120/v2/devices]
[12/14/2020, 8:26:58 PM] [Bond] Request (185d990e38e94000) [get http://192.168.0.120/v2/sys/version]
---
[Bond] Response (185d990e38e8c000) [get http://192.168.0.119/v2/devices] - {"1":{"_":"3b4962fb"},"_":"931831bd"}
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e38e90000) [get http://192.168.0.119/v2/sys/version] - {"target":"breck-northstar","fw_ver":"v2.15","fw_date":"Thu Oct 22 18:33:56 UTC 2020","uptime_s":106347,"make":"Unbranded","model":"MR196W","branding_profile":"KOF_MR196W","bondid":"KNKSADE42861","upgrade_http":true,"api":2,"_":"e3fdc04c"}
[12/14/2020, 8:26:59 PM] [Bond] 
****** Bond Info *******
 bondId: KNKSADE42861
 FW: v2.15
 API: v2
 Make: MR196W
 Model: MR196W
************************
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e38e7c000) [get http://192.168.0.118/v2/devices] - {"1":{"_":"bdcea4dd"},"_":"af164203"}
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e38e90001) [get http://192.168.0.120/v2/devices] - {"1":{"_":"56eed881"},"_":"36dbe6b2"}
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e38e88000) [get http://192.168.0.118/v2/sys/version] - {"target":"breck-northstar","fw_ver":"v2.15","fw_date":"Thu Oct 22 18:33:56 UTC 2020","uptime_s":106346,"make":"Chungear Industrial Co., Ltd.","model":"MR196W","branding_profile":"KOF_KNKSA","bondid":"KNKSACG91580","upgrade_http":true,"api":2,"_":"c1e86b0e"}
[12/14/2020, 8:26:59 PM] [Bond] 
****** Bond Info *******
 bondId: KNKSACG91580
 FW: v2.15
 API: v2
 Make: MR196W
 Model: MR196W
************************
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e38e94000) [get http://192.168.0.120/v2/sys/version] - {"target":"breck-northstar","fw_ver":"v2.15","fw_date":"Thu Oct 22 18:33:56 UTC 2020","uptime_s":106073,"make":"Unbranded","model":"MR196W","branding_profile":"KOF_MR196W","bondid":"KNKSADE41829","upgrade_http":true,"api":2,"_":"e438dfbc"}
[12/14/2020, 8:26:59 PM] [Bond] 
****** Bond Info *******
 bondId: KNKSADE41829
 FW: v2.15
 API: v2
 Make: MR196W
 Model: MR196W
************************
[12/14/2020, 8:26:59 PM] [Bond] 3 cached accessories were loaded
[12/14/2020, 8:26:59 PM] [Bond] Getting devices for this Bond (KNKSACG91580)...
[12/14/2020, 8:26:59 PM] [Bond] 1 devices were found on this Bond (KNKSACG91580).
[12/14/2020, 8:26:59 PM] [Bond] No new devices to add for this Bond (KNKSACG91580).
[12/14/2020, 8:26:59 PM] [Bond] Getting devices for this Bond (KNKSADE42861)...
[12/14/2020, 8:26:59 PM] [Bond] 1 devices were found on this Bond (KNKSADE42861).
[12/14/2020, 8:26:59 PM] [Bond] No new devices to add for this Bond (KNKSADE42861).
[12/14/2020, 8:26:59 PM] [Bond] Getting devices for this Bond (KNKSADE41829)...
[12/14/2020, 8:26:59 PM] [Bond] 1 devices were found on this Bond (KNKSADE41829).
[12/14/2020, 8:26:59 PM] [Bond] No new devices to add for this Bond (KNKSADE41829).
[12/14/2020, 8:26:59 PM] [Bond] UDP message sent to 192.168.0.118:30007
[12/14/2020, 8:26:59 PM] [Bond] UDP message sent to 192.168.0.119:30007
[12/14/2020, 8:26:59 PM] [Bond] UDP message sent to 192.168.0.120:30007
[12/14/2020, 8:26:59 PM] [Bond] UDP Message received from 192.168.0.118:30007 - {"B":"KNKSACG91580"}
[12/14/2020, 8:26:59 PM] [Bond] UDP Message received from 192.168.0.120:30007 - {"B":"KNKSADE41829"}
[12/14/2020, 8:26:59 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861"}
[12/14/2020, 8:26:59 PM] [Bond] Configuring Accessory: Master Bedroom MBR Fan
[12/14/2020, 8:26:59 PM] [Bond] [MBR Fan] actions: TurnOn,TurnOff,TogglePower,SetSpeed,IncreaseSpeed,DecreaseSpeed,TurnLightOn,TurnLightOff,ToggleLight,SetBrightness,IncreaseBrightness,DecreaseBrightness,CycleBrightness,SetTimer,BreezeOn,BreezeOff,SetBreeze
[12/14/2020, 8:26:59 PM] [Bond] Request (185d990e3a11c000) [get http://192.168.0.118/v2/devices/1/state]
[12/14/2020, 8:26:59 PM] [Bond] Configuring Accessory: Dining Room DR Fan
[12/14/2020, 8:26:59 PM] [Bond] [DR Fan] actions: TurnOn,TurnOff,TogglePower,SetSpeed,IncreaseSpeed,DecreaseSpeed,TurnLightOn,TurnLightOff,ToggleLight,SetBrightness,IncreaseBrightness,DecreaseBrightness,CycleBrightness,SetTimer,BreezeOn,BreezeOff,SetBreeze
[12/14/2020, 8:26:59 PM] [Bond] Request (185d990e3a128000) [get http://192.168.0.119/v2/devices/1/state]
[12/14/2020, 8:26:59 PM] [Bond] Configuring Accessory: Living Room LV Fan
[12/14/2020, 8:26:59 PM] [Bond] [LV Fan] actions: TurnOn,TurnOff,TogglePower,SetSpeed,IncreaseSpeed,DecreaseSpeed,TurnLightOn,TurnLightOff,ToggleLight,SetBrightness,IncreaseBrightness,DecreaseBrightness,CycleBrightness,SetTimer,BreezeOn,BreezeOff,SetBreeze
[12/14/2020, 8:26:59 PM] [Bond] Request (185d990e3a134000) [get http://192.168.0.120/v2/devices/1/state]
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e3a11c000) [get http://192.168.0.118/v2/devices/1/state] - {"power":0,"speed":1,"light":0,"brightness":100,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"075504fa"}
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e3a134000) [get http://192.168.0.120/v2/devices/1/state] - {"power":0,"speed":1,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"e3d36cd3"}
[12/14/2020, 8:26:59 PM] [Bond] Response (185d990e3a128000) [get http://192.168.0.119/v2/devices/1/state] - {"power":0,"speed":1,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"5d0c04fb"}
---
[12/14/2020, 8:27:59 PM] [Bond] UDP message sent to 192.168.0.118:30007
[12/14/2020, 8:27:59 PM] [Bond] UDP message sent to 192.168.0.119:30007
[12/14/2020, 8:27:59 PM] [Bond] UDP message sent to 192.168.0.120:30007
[12/14/2020, 8:27:59 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861"}
[12/14/2020, 8:27:59 PM] [Bond] UDP Message received from 192.168.0.118:30007 - {"B":"KNKSACG91580"}
[12/14/2020, 8:27:59 PM] [Bond] UDP Message received from 192.168.0.120:30007 - {"B":"KNKSADE41829"}
---
Turning the fan on
[12/14/2020, 8:30:06 PM] [Bond] Request (185d990ef0b24000) [put http://192.168.0.119/v2/devices/1/actions/SetSpeed] - body: {"argument":4}
[12/14/2020, 8:30:06 PM] [Bond] Request (185d990ef0b30000) [put http://192.168.0.119/v2/devices/1/actions/TurnOn]
[12/14/2020, 8:30:06 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861","t":"devices/1/state","i":"050000d888c70994","f":100,"s":200,"m":0,"x":"bond","b":{"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"5c32f6e1"}}
[12/14/2020, 8:30:06 PM] [Bond] [DR Fan] Received new state: {"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"5c32f6e1"}
[12/14/2020, 8:30:06 PM] [Bond] Response (185d990ef0b24000) [put http://192.168.0.119/v2/devices/1/actions/SetSpeed] - {"argument":4,"_":"dfb92807"}
[12/14/2020, 8:30:06 PM] [Bond] [DR Fan] Set fan speed: 4
[12/14/2020, 8:30:06 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861","t":"devices/1/state","i":"050000d9ce8c8d90","f":100,"s":200,"m":0,"x":"bond","b":{"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"291c60af"}}
[12/14/2020, 8:30:06 PM] [Bond] [DR Fan] Received new state: {"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"291c60af"}
[12/14/2020, 8:30:07 PM] [Bond] Response (185d990ef0b30000) [put http://192.168.0.119/v2/devices/1/actions/TurnOn] - {"_":"3ba53940"}
[12/14/2020, 8:30:07 PM] [Bond] [DR Fan] Set fan power: true
[12/14/2020, 8:30:08 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861","t":"devices/1/state","i":"0500000eeda1c981","f":100,"s":200,"m":0,"x":"bond","b":{"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"fdb58a40"}}
[12/14/2020, 8:30:08 PM] [Bond] [DR Fan] Received new state: {"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"fdb58a40"}
---
Turning the light on
[12/14/2020, 8:30:54 PM] [Bond] Request (185d990f1fa0c000) [put http://192.168.0.119/v2/devices/1/actions/ToggleLight]
[12/14/2020, 8:30:55 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861","t":"devices/1/state","i":"0500000b89eeed46","f":100,"s":200,"m":0,"x":"bond","b":{"power":1,"speed":4,"light":1,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"c641e979"}}
[12/14/2020, 8:30:55 PM] [Bond] [DR Fan] Received new state: {"power":1,"speed":4,"light":1,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"c641e979"}
[12/14/2020, 8:30:55 PM] [Bond] Response (185d990f1fa0c000) [put http://192.168.0.119/v2/devices/1/actions/ToggleLight] - {"_":"f0d0d6b6"}
[12/14/2020, 8:30:55 PM] [Bond] [DR Fan] Set light power: true
[12/14/2020, 8:30:56 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861","t":"devices/1/state","i":"0500009f91318d04","f":100,"s":200,"m":0,"x":"bond","b":{"power":1,"speed":4,"light":1,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"43095f9c"}}
[12/14/2020, 8:30:56 PM] [Bond] [DR Fan] Received new state: {"power":1,"speed":4,"light":1,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"43095f9c"}
---
Turning the light off
[12/14/2020, 8:32:22 PM] [Bond] Request (185d990f757e0000) [put http://192.168.0.119/v2/devices/1/actions/ToggleLight]
[12/14/2020, 8:32:23 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861","t":"devices/1/state","i":"05000089d032102a","f":100,"s":200,"m":0,"x":"bond","b":{"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"e421a1b6"}}
[12/14/2020, 8:32:23 PM] [Bond] [DR Fan] Received new state: {"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"e421a1b6"}
[12/14/2020, 8:32:23 PM] [Bond] Response (185d990f757e0000) [put http://192.168.0.119/v2/devices/1/actions/ToggleLight] - {"_":"f0d0d6b6"}
[12/14/2020, 8:32:23 PM] [Bond] [DR Fan] Set light power: false
[12/14/2020, 8:32:24 PM] [Bond] UDP Message received from 192.168.0.119:30007 - {"B":"KNKSADE42861","t":"devices/1/state","i":"05000066636347ab","f":100,"s":200,"m":0,"x":"bond","b":{"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"1a542fb1"}}
[12/14/2020, 8:32:24 PM] [Bond] [DR Fan] Received new state: {"power":1,"speed":4,"light":0,"brightness":50,"brightness_cycle_phase":-1,"timer":0,"breeze":[0,50,50],"_":"1a542fb1"}

But when trying to change the light intensity, absolutely nothing appears in the log. I move the slider in HomeKit, but it has no action., until I reach the value "0", and then the light turns off.

Please note the value "50" for brightness was selected through the "Bond" app. Selecting the intensity in the Bond app works fine.

Let me know if you need anything else.

**Comment by aarons22** [comment]
This was helpful! I may have stumbled onto the issue. Turns out I was trying to set the fan speed instead of the brightness (probably from when I did copy/paste from the fan speed feature as a starting point 🤦 ). 

Give it a shot in [3.2.2](https://github.com/aarons22/homebridge-bond/releases/tag/3.2.2) and let me know what you see!

**Comment by Silmar-99** [comment]
Unfortunately, 3.2.2 didn't resolved the issue. On/Off is still working, but the Dimmer is still not, with no entry in the log when using it...

**Comment by randymassey** [comment]
Hi, I can confirm the same in testing 322.  It’s still not dimming for me either.

Every other function for the fan works, speeds of the fan, reversing direction, & turning the light on & off will work, but the dimming function yields no activity, tho you can adjust the slider.  In basic logging there will be no log data when you change brightness & there is no flashing on the BOND that commands are being issued to the fan lamp for the light dimming.

FYI..  Randy

**Comment by methnen** [comment]
Just another voice to the gallery.

Recently replaced my older (regular) Artemis fan with a Smart by Bond one that I got used, everything works great and I'm so glad to finally be over the whole light state issue (is it on or is it off?!!!!) everything seems to be working perfectly with Homebridge EXCEPT for the light dimming feature which is either off or 100% regardless of what you set it at in Homebridge or HomeKit.

Happy to try and capture logs or whatever if there's something I can do to help.

**Comment by aarons22** [comment]
Okay, I was finally able to reproduce locally by faking a few things with my fan, and I found the issue. I was listening for brightness changes on the standard light accessory, but not on the fan light accessory. It explains why I was so confused because I thought I was doing everything right, but turns out, I needed to do the same functionality in two places!

TL;DR:  [3.2.3](https://github.com/aarons22/homebridge-bond/releases/tag/3.2.3) should have this issue fixed.

**Comment by Silmar-99** [comment]
Aaron,
I will let the others confirm as well, but I think you got it right this time. I tried on all three of my fans and the light is now dimming at it should. Bonus, when changing the value with the remote provided with the fan, the value also change in sync (with a delay of a second or so, but anyway, it works!).
Thanks for your hard work on this one!

**Comment by methnen** [comment]
Same, all good on the western front.

**Comment by aarons22** [comment]
Great! Glad we figured this one out.
